libc/resolve_path.c

```diff
@@ -475,6 +475,47 @@ TEST(resolve_path, symlink_rename)
 }
 
 
+TEST(resolve_path, symlink_enametoolong)
```

Comment:
That's not only about ENAMETOOLONG, rather a special case, which may raise ENAMETOOLONG errno. Above that we have test case named errno in this test, which also checks ENAMETOOLONG (it could be misleading that it's not in enametoolong).

```suggestion
TEST(resolve_path, symlink_long_resolution)
```